app/models/hud_reports/report_client_base.rb

```diff
@@ -13,56 +13,86 @@ class ReportClientBase < GrdaWarehouseBase
 
     has_many :report_clients, as: :universe_membership, dependent: :destroy
 
-    def display_value(col, pii_policy:, include_content_tag: true)
+    def display_value(col, pii_policy:, include_content_tag: true, cell_val: nil, calculate_cell: true)
+      cell_val = fetch_cell_value(col) if calculate_cell
+
+      return format_complex_value(col, cell_val, pii_policy, include_content_tag) if cell_val.is_a?(Array) || cell_val.is_a?(Hash)
+      return format_boolean(cell_val, include_content_tag) if cell_val.in?([true, false])
+
+      transform_value(col.to_s, cell_val, pii_policy)
+    end
+
+    private
+
+    def fetch_cell_value(col)
       # We are expecting some columns to come through with a dot notation.
       # This will separate the object and method in order to call the appropriate column/method
       # while similarly allowing the method to be called if the column is not passed using dot notation.
-      cell = col.to_s.split('.').inject(self) do |obj, method|
-        break nil if obj.nil? # stop chaining if nil is reached
-
+      col.to_s.split('.').inject(self) do |obj, method|
         if obj.class.column_names.include?(method)
           obj[method]
         else
           obj.send(method)
         end
       end
+    end
 
-      return ActionController::Base.helpers.content_tag(:pre, JSON.pretty_generate(cell)) if include_content_tag && (cell.is_a?(Array) || cell.is_a?(Hash))
-      return Reports::ModelApplicationHelper.new.yes_no(cell, include_content_tag: include_content_tag) if cell.in?([true, false])
+    def format_complex_value(col, value, pii_policy, include_content_tag)
+      if value.is_a?(Array)
+        # For Arrays, calculate each array element's value using the column name for the array
+        value.map { |item| display_value(col, pii_policy: pii_policy, include_content_tag: include_content_tag, cell_val: item, calculate_cell: false) }
+      elsif value.is_a?(Hash)
+        # For Hashes, calculate each entry's value using each entry's key as the column name
+        value.each do |k, v|
+          value[k] = display_value(k.to_s, pii_policy: pii_policy, include_content_tag: include_content_tag, cell_val: v, calculate_cell: false)
+        end
+      end
+
+      if include_content_tag
+        ActionController::Base.helpers.content_tag(:pre, JSON.pretty_generate(value))
+      else
+        value
+      end
```

Comment:
What do you think about:
```suggestion
      return ActionController::Base.helpers.content_tag(:pre, JSON.pretty_generate(value)) if include_content_tag

      value
```